src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -26,6 +31,221 @@
+    # Optimization note: this step is done outside of main loop above since
+    # it requires fetching block information from DBS which may not be required
+    # if we do not have such blocks.
+    if len(blocksToUpdate) > 0:
+        logger.info("Adding %s blocks from MSPileup which are not present in pileupconf.json",
+                    len(blocksToUpdate))
+        binfo = getBlockInfo(blocksToUpdate)
+        for puType in returnDict.keys():
+            for blockName in blocksToUpdate:
+                # update block record in-place
+                record = returnDict[puType][blockName]
+                record["FileList"] = binfo[blockName]["files"]
+                record["NumberOfEvents"] = binfo[blockName]["nevents"]
+    return returnDict
+
+
+def getBlockInfo(blockNames):
+    """
+    Fetch block information details, file list and number of events, from DBS
+    server.
+    :param blockNames: list of block names
+    :return dict: dictionary of {block: {"files": file_list, "nevents": number_events}, ...}
+    """
+    # TODO: the logic of this function should be implemented concurrently:
+    # - for every given block get information from DBS: list of files and number of events in a block
+    # - please use pycurl_manager.py module and the following logic:
+#    urls = []
+#    for blk in blockNames:
+#        # need to encode block name properly
+#        block = urllib.parse.quote_plus(blk)
+#        url = f"https://cmsweb-prod.cern.ch/dbs/prod/global/DBSReader/datasetsummary?block_name={block}"
+#        urls.append(url)
+#    # place concurrent calls to DBS
+#    results = getdata(urls, ckey, cert, cookie=cookie)
+#    # parse output of getdata in some form
+#    for row in resuls:
+#        blockInfo[blk] = ...
+
+    # so far we put placeholder codebase which does nothing
+    fileList = []
+    numEvents = 0
+    blockInfo = {}
+    for blk in blockNames:
+        blockInfo[blk] = {"files": fileList, "nevents": numEvents}
+    return blockInfo
+
+
+def writePileupJson(tfile, jdict, dest=None):
+    """
+    Write pileup JSON sandbox files back to file system
+    :param tfile: tar ball file name
+    :param jdict: JSON sandbox dictionary in form: {"path-to-json1": jdoc1, "path-to-json2": jdoc2}
+    :param dest: optional destination parameter to write final tar ball (use for unit tests)
+    :return: nothing
+    """
+    bname = os.path.basename(tfile)
+    dname = os.path.dirname(tfile)
+    ofile = f"{dname}/new-{bname}"
+    with tempfile.TemporaryDirectory() as tmpDir:
+        # extract tar ball content into temporary directory
+        with tarfile.open(tfile, tarMode(tfile, 'r'), encoding='utf-8') as tar:
+            tar.extractall(path=tmpDir)
+        # overwrite json sanbox file in temporary directory
+        for jname in jdict:
+            fname = os.path.join(tmpDir, jname)
+            fstat = os.stat(fname)
+            with open(fname, 'w', encoding='utf-8') as ostream:
```

Comment:
The appropriate log message is done outside of this function call since this is stand-alone function without logger, see https://github.com/dmwm/WMCore/pull/11884/files/89b9695bb0d5c1323cf12c10da9669a2870263b5#diff-e42968b3836140494b36bc2f6a339ba46c737f8e6de3117c8f81e5cec97320a9R436

If you want log message within this function we need to pass logger as input parameter. Please let me know your preference and I can adjust accordingly.

Reply:
I don't see it logging which relative files are being updated in the tarball. I think it would be important. You can either print all of that before calling this function, or pass in a logger object and log it here.

Reply:
ok, added logger and appropriate message inside this function